### PR TITLE
Repave clarifications

### PR DESCRIPTION
**Pull request by cloudmaniac**
## Describe the Change

This PR clarifies the different types of repave that can happen in a cluster.

### CODE REVIEW

File: docs/docs-content/clusters/cluster-management/node-pool.md [hunk 1]
```diff
@@ -18,10 +18,17 @@ Ensure you exercise caution when modifying node pools. We recommend creating a [
...
+In Kubernetes, the term "repave" refers to the process of replacing a node with a new node. [Repaving](/glossary-all#repavement) is a common practice in Kubernetes to ensure that nodes are deployed with the latest version of the operating system and Kubernetes. Repaving is also used to replace nodes that are unhealthy or have failed.
 
-The ability to configure the repave time interval for all node pools except the master pool. The default repave time interval is 0 seconds. You can configure the node repave time interval during the cluster creation process or after the cluster is created. To modify the repave time interval after the cluster is created, follow the [Change a Node Pool](#changeanodepool) instructions below.
+Different types of repaving operations may occur, contingent upon the actions that trigger them:
 
+* Control plane repave: takes place when specific alterations are made to the Kubernetes configuration, such as changing the apiServer specification, or when there are changes in the hardware specifications of the master nodes (e.g., during a scale-up operation or when changing from an instance to another). Only control plane nodes are repaved, one by one.
```
[the remark this answers is not in the document]
Comment (antongisli) [reply docs/docs-content/clusters/cluster-management/node-pool.md:25]: do you mean "instance type"? Last sentence can be "Control plane nodes will be repaved sequentially."

[the remark this answers is not in the document]
Comment (antongisli) [reply docs/docs-content/clusters/cluster-management/node-pool.md:25]: also, avoid word "master", we can use control plane nodes as the term?

**Review by kreeuwijk — APPROVED** [review]
LGTM

**Comment by cloudmaniac** [comment]
@karl-cardenas-coding This can be merged if you're good with the language

File: docs/docs-content/clusters/cluster-management/node-pool.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ sidebar_position: 190
 tags: ["clusters", "cluster management"]
 ---
 
-A node pool is a group of nodes within a cluster that all have the same configuration. Node pools allow you to create pools of nodes that can be used for different workloads. For example, you can create a node pool for your production workloads and another node pool for your development workloads. You can update node pools for active clusters or create a new node pool for the cluster. 
+A node pool is a group of nodes within a cluster that all have the same configuration. Node pools allow you to create pools of nodes that can be used for different workloads. For example, you can create a node pool for your production workloads and another node pool for your development workloads. You can update node pools for active clusters or create a new node pool for the cluster.
```
Comment (ritawatson) [inline docs/docs-content/clusters/cluster-management/node-pool.md:10]:
```suggestion
A node pool is a group of nodes within a cluster that all have the same configuration. You can use node pools for different workloads. For example, you can create a node pool for your production workloads and another for your development workloads. You can update node pools for active clusters or create a new one for the cluster.
```

Comment (ritawatson) [reply docs/docs-content/clusters/cluster-management/node-pool.md:10]: Reduced the number of occurrences of "node pool".

File: docs/docs-content/clusters/cluster-management/node-pool.md [hunk 3]
```diff
@@ -18,10 +18,17 @@ Ensure you exercise caution when modifying node pools. We recommend creating a [
...
+In Kubernetes, the term "repave" refers to the process of replacing a node with a new node. [Repaving](/glossary-all#repavement) is a common practice in Kubernetes to ensure that nodes are deployed with the latest version of the operating system and Kubernetes. Repaving is also used to replace nodes that are unhealthy or have failed.
 
-The ability to configure the repave time interval for all node pools except the master pool. The default repave time interval is 0 seconds. You can configure the node repave time interval during the cluster creation process or after the cluster is created. To modify the repave time interval after the cluster is created, follow the [Change a Node Pool](#changeanodepool) instructions below.
+Different types of repaving operations may occur, contingent upon the actions that trigger them:
```
Comment (ritawatson) [inline docs/docs-content/clusters/cluster-management/node-pool.md:23]:
```suggestion
Different types of repaving operations may occur, depending on what causes them:
```

File: docs/docs-content/clusters/cluster-management/node-pool.md [hunk 4]
```diff
@@ -18,10 +18,17 @@ Ensure you exercise caution when modifying node pools. We recommend creating a [
...
 
+* Control plane repave: takes place when specific alterations are made to the Kubernetes configuration, such as changing the apiServer specification, or when there are changes in the hardware specifications of the control plane nodes (for example during a scale-up operation or when changing from an instance type to another). Control plane nodes will be repaved sequentially.
```
Comment (ritawatson) [inline docs/docs-content/clusters/cluster-management/node-pool.md:25]:
```suggestion
* **Control plane repave**: This takes place when certain changes are made to the Kubernetes configuration, such as changing the apiServer specification. This type of repave also occurs when there are changes in the hardware specifications of the control plane nodes, such as during a scale-up operation or when changing from one instance type to another. Control plane nodes are repaved sequentially.
```

File: docs/docs-content/clusters/cluster-management/node-pool.md [hunk 5]
```diff
@@ -18,10 +18,17 @@ Ensure you exercise caution when modifying node pools. We recommend creating a [
...
+
+* Worker node pool repave: takes place when alterations to the specification of a particular node pool render the existing nodes incompatible with the pool's criteria. For instance, changing the hardware specifications of a worker pool. In this case, nodes within the affected pool are sequentially replaced with new nodes that meet the updated pool specifications.
```
Comment (ritawatson) [inline docs/docs-content/clusters/cluster-management/node-pool.md:27]:
```suggestion
* **Worker node pool repave**: This happens when changes to a node pool's specifications render the existing nodes incompatible with the pool's criteria. For instance, changing the hardware specifications of a worker pool. Nodes within the affected pool are sequentially replaced with new nodes that meet the updated specifications.
```

File: docs/docs-content/clusters/cluster-management/node-pool.md [hunk 6]
```diff
@@ -18,10 +18,17 @@ Ensure you exercise caution when modifying node pools. We recommend creating a [
...
+
+* Full cluster repave: occurs if any changes are made to the OS layer or if there are modifications to the Kubernetes layer that impact all nodes (for example when upgrading to a different version of Kubernetes). In a full cluster repaving, all nodes across all pools, including the control plane, undergo repaving one by one, with control plane nodes taking precedence, followed by worker pool nodes.
```
Comment (ritawatson) [inline docs/docs-content/clusters/cluster-management/node-pool.md:29]:
```suggestion
* **Full cluster repave**: This occurs if changes are made to the Operating System (OS) layer or if modifications to the Kubernetes layer impact all nodes, such as when upgrading to a different Kubernetes version. All nodes across all pools are sequentially repaved starting with the control plane.
```

File: docs/docs-content/clusters/cluster-management/node-pool.md [hunk 7]
```diff
@@ -18,10 +18,17 @@ Ensure you exercise caution when modifying node pools. We recommend creating a [
...
+
+You can customize the repave time interval for all node pools, with the exception of the master pool. The default repave time interval is 0 seconds. You have the flexibility to adjust the node repave time interval either during the cluster creation or after the cluster has been created. If you need to modify the repave time interval post-cluster creation, follow the [Change a Node Pool](#changeanodepool) instructions below.
```
Comment (ritawatson) [inline docs/docs-content/clusters/cluster-management/node-pool.md:31]:
```suggestion
You can customize the repave time interval for all node pools except the master pool. The default repave time interval is 0 seconds. You can adjust the node repave time interval during or after cluster creation. If you need to modify the repave time interval post-cluster creation, follow the [Change a Node Pool](#changeanodepool) instructions below.
```

Comment (ritawatson) [reply docs/docs-content/clusters/cluster-management/node-pool.md:31]: Line 31 uses the term "master". In this case, we could say, "You can customize the repave time interval for all node pools except the control plane node pool."

File: docs/docs-content/glossary-all.md [hunk 8]
```diff
@@ -158,7 +158,7 @@ Palette maintains a public pack registry containing various [packs](#pack) that
 
 ## Repavement
 
-Repavement is the process of replacing a Kubernetes node with a new one. This is typically done when a node is unhealthy or needs to be upgraded. The process involves draining the node, or in other words, migrating active workloads to another healthy node, and removing it from the cluster. A new node is created and configured with the same settings as the old node and added back to the cluster. The process is fully automated and does not require any manual intervention.
+Repavement is the process of replacing a Kubernetes node with a new one. This is typically done when a node is unhealthy or needs to be upgraded. The process involves draining the node, or in other words, migrating active workloads to another healthy node, and removing it from the [node pool](/clusters/cluster-management/node-pool/#repave-behavior-and-configuration). A new node is created and configured with the same settings as the old node and added back to the pool. The process is fully automated and does not require any manual intervention.
```
Comment (ritawatson) [inline docs/docs-content/glossary-all.md:161]:
```suggestion
Repavement is the process of replacing a Kubernetes node with a new one. This is typically done when a node is unhealthy or needs to be upgraded. The process involves migrating active workloads to another healthy node, and removing it from the [node pool](/clusters/cluster-management/node-pool/#repave-behavior-and-configuration). This is referred to as draining the node. A new node is created and configured with the same settings as the old node and added back to the pool. The process is fully automated and does not require manual intervention.

```

**Review by ritawatson — APPROVED** [review]
There are some comments to address before this PR can be merged.

**Review by cloudmaniac — COMMENTED** [review]
Accepted suggestions

**Review by karl-cardenas-coding — APPROVED** [review]

**Pull request merged**